magicctapipe/scripts/lst1_magic/semi_automatic_scripts/DL2_to_DL3.py

```diff
@@ -0,0 +1,314 @@
+
+import joblib
+import numpy as np
+import pandas as pd
+import yaml
+
+from magicctapipe import __version__
+from magicctapipe.io import resource_file
+from magicctapipe.scripts.lst1_magic.semi_automatic_scripts.clusters import (
+    rc_lines,
+    slurm_lines,
+)
+
+__all__ = ["configuration_DL3", "DL2_to_DL3"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def configuration_DL3(target_dir, source_name, config_file, ra, dec):
+    """
+    This function creates the configuration file needed for the DL2 to DL3 conversion
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    source_name : str
+        Source name
+    config_file : str
+        Path to MCP configuration file (e.g., resources/config.yaml)
+    ra : float
+        Source RA
+    dec : float
+        Source Dec
+    """
+
+    if config_file == "":
+        config_file = resource_file("config.yaml")
+
+    with open(
+        config_file, "rb"
+    ) as fc:  # "rb" mode opens the file in binary format for reading
+        config_dict = yaml.safe_load(fc)
+    DL3_config = config_dict["dl2_to_dl3"]
+    DL3_config["source_name"] = source_name
+    DL3_config["source_ra"] = f"{ra} deg"
+    DL3_config["source_dec"] = f"{dec} deg"
+    conf = {
+        "mc_tel_ids": config_dict["mc_tel_ids"],
+        "dl2_to_dl3": DL3_config,
+    }
+
+    conf_dir = f"{target_dir}/v{__version__}/{source_name}"
+    os.makedirs(conf_dir, exist_ok=True)
+
+    file_name = f"{conf_dir}/config_DL3.yaml"
+
+    with open(file_name, "w") as f:
+
+        yaml.dump(conf, f, default_flow_style=False)
+
+
+def DL2_to_DL3(
+    target_dir,
+    source,
+    env_name,
+    IRF_dir,
+    df_LST,
+    cluster,
+    MC_v,
+    version,
+    nice,
+    IRF_theta_cuts_type,
+    LST_date,
+):
+    """
+    This function creates the bash scripts to run lst1_magic_dl2_to_dl3.py on the real data.
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    source : str
+        Source name
+    env_name : str
+        Conda enviroment name
+    IRF_dir : str
+        Path to the IRFs
+    df_LST : :class:`pandas.DataFrame`
+        Dataframe collecting the LST1 runs (produced by the create_LST_table script)
+    cluster : str
+        Cluster system
+    MC_v : str
+        Version of MC processing
+    version : str
+        Version of the input (stereo subruns) data
+    nice : int or None
+        Job priority
+    IRF_theta_cuts_type : str
+        Type of IRFS (global/dynamic)
+    LST_date : list
+        List of the dates to be processed (from list_from_h5)
+    """
+    if cluster != "SLURM":
+        logger.warning(
+            "Automatic processing not implemented for the cluster indicated in the config file"
+        )
+        return
+
+    # Loop over all nights
+    outdir = f"{target_dir}/v{__version__}/{source}/DL3/logs"
```

Comment:
this changes the output scheme removing the date. It has some advantage that you can calculate the gammapy databases from the whole source, but it is a minor thing.
The disadvantage is that you cannot easily select the data via date, that can be helpful not only for analyzers, but also for automatic processing if we run it parts by parts and make some mistake